docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,153 @@
+
+Assumptions & Out-of-scope
+--------------------------
+
+-   This is only for platforms that currently support split tunnel - Windows, Linux, and Android. We will not build split tunneling into macOS or iOS as part of this work.
+-   Notifications refer to system-level notifications. There are no additions to the messages tab considered in this tech spec.
+
+Open Questions
+--------------
+
+-   What do we use to canonically track apps we may want to exclude? Is there a bundle ID? The name of app (fragile)? Something else? Do we use different IDs across platforms?
+-   Any easy ways to categorize games? (See [VPN-4433](https://mozilla-hub.atlassian.net/browse/VPN-4433).)
+-   How will we decide which apps are initially on the suggested list?
+-   Does the suggested list just include specific apps, or can it include entire directories (like well-known Windows game directories) and/or allow wildcard matching?
+-   What is the process for adding an app to list, in light of potential security/privacy concerns when adding malicious apps?
+
+Implementation (estimated 36-57 points total)
+---------------------------------------------------
+
+This work is broken into four sections, with the bulk of code coming in the middle two sections. Those two sections should each be built behind their own separate feature flag, though we'd expose both parts together via Nimbus.
+
+### Prerequisites (15-26 points total)
+
+-   (3-5 points) Answer the open questions above
+-   (5-8 points) Create a well-structured file (for our own research, as well as one users can add to via GitHub pull requests on the client repo) w/ apps to exclude (and update the README in the repo to explain this new type of contribution). These docs will live in our repo, and be bundled with the app. Add tests to ensure the doc's format integrity is maintained. This format should be designed to allow new platforms (like websites) to be added without breaking anything.
+    - The document format will be a single JSON file for all platforms. Within each platform there will be an array of IDs for the suggested apps. These will each be an array, and we'll only look at the first item in the array - if a second string exists in the array will be considered comments (as JSON doesn't have the concept of comments). (This is preferred to keys/values for binary/comments compactness/legibility.)
+    ```
+    {
+        "android":
+            [
+                ["my.app.identifier", "an optional comment"],
+                ["my.other.app.identifier"]
+            ],
+        "linux":
+            [
+                ["whatever.identifier.linux.uses", "comments are still optional"]
+            ]
+    }
+```
+    -   Additions to the list should be given the same amount of thought that adding new dependencies to our repo are given. That is, full research by our engineers to ensure nothing malicious is being brought in. And erring on the side of not including any.
```

Comment:
nit: Consider calling out that defining the actual steps / requirements for approval of new apps to the list is out of scope for this document, but will be done is a follow up workflow. I think that would be important, to be clear that we don't plan to handwave this -- "full research by our engineers to ensure nothing malicious is being brought in" is quite vague, which is fine for this document, but not fine for the feature itself.

Reply:
👍🏻